### PR TITLE
Create a printing module

### PR DESCRIPTION
**Pull request by luksan47**
Added base of Controller and Model.
Left some tasks open for newcomers.

### CODE REVIEW

File: app/Http/Controllers/PrintController.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+use App\PrintAccount;
+
+class PrintController extends Controller
+{
+    public function index() {
+        return view('print.app');
+    }
+
+    public function modify_balance(Request $request) {
+        $printAccount = User::find($request->user_id)->printAccount;
```
Comment (vendi95) [inline app/Http/Controllers/PrintController.php]: Use [findOrFail ](https://laravel.com/docs/5.8/eloquent#retrieving-single-models) to handle invalid ids.

File: app/Http/Controllers/PrintController.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        $printAccount->balance += $request->balance;
+        $printAccount->save();
```
Comment (vendi95) [inline app/Http/Controllers/PrintController.php]: Should use [increment](https://laravel.com/docs/5.8/queries#increment-and-decrement) for atomicity.

File: config/database.php [hunk 3]
```diff
@@ -40,7 +40,7 @@
             'url' => env('DATABASE_URL'),
             'database' => env('DB_DATABASE', database_path('database.sqlite')),
             'prefix' => '',
-            'foreign_key_constraints' => env('DB_FOREIGN_KEYS', true),
+            'foreign_key_constraints' => true,
```
Comment (vendi95) [inline config/database.php]: I think this should not be changed here, edit DB_FOREIGN_KEYS constant in .env file instead.

File: resources/views/print/modify.blade.php [hunk 4]
```diff
@@ -1,6 +1,11 @@
 <div class="card">
     <div class="card-header">@lang('print.modify')</div>
     <div class="card-body">
-       
+        <form method="POST" action="/print/modify_balance">
```
Comment (vendi95) [inline resources/views/print/modify.blade.php]: Should use [route ](https://laravel.com/docs/5.8/routing#named-routes) function with the named route to get action URL.

File: routes/web.php [hunk 5]
```diff
@@ -17,12 +17,8 @@
...
-
-// TODO: move to controller
-Route::get('/print', function () {
-    return view('print.app');
-});
\ No newline at end of file
+Route::middleware(['auth'])->group(function () {
+    Route::get('/home', 'HomeController@index')->name('home');
+    Route::get('/print', 'PrintController@index')->name('print');
+    Route::post('/print/modify_balance', 'PrintController@modify_balance');
```
Comment (vendi95) [inline routes/web.php]: No authorization needed?

File: config/database.php (hunk 3 above)
Comment (luksan47) [reply config/database.php]: I would enforce it in the production environment to ensure that foreign key constraints are always applied. What could be the benefits of keeping this option in the .env file?

File: routes/web.php (hunk 5 above)
Comment (luksan47) [reply routes/web.php]: They are grouped under the 'auth' middleware which enforces authorization.

File: resources/views/print/modify.blade.php (hunk 4 above)
Comment (luksan47) [reply resources/views/print/modify.blade.php]: Done.

File: app/Http/Controllers/PrintController.php (hunk 2 above)
Comment (luksan47) [reply app/Http/Controllers/PrintController.php]: Done.

File: routes/web.php (hunk 5 above)
Comment (vendi95) [reply routes/web.php]: OK. I was thinking about the checking of admin permissions for modifying print balance. But as I see, permission management is not part of this issue and will only be added later.

File: app/Http/Controllers/PrintController.php (hunk 1 above)
Comment (luksan47) [reply app/Http/Controllers/PrintController.php]: Done.

File: config/database.php (hunk 3 above)
Comment (vendi95) [reply config/database.php]: I do not think we would like to use sqlite in production :) With this change, you can not even turn it off for your local developer environment without code changes. But I do not insist on this.

Comment (luksan47) [reply config/database.php]: Hm, fine, we can leave it more flexible.

**Pull request merged**